modules/compute/variables.tf

```
@@ -0,0 +1,29 @@
+variable "location" {
+  type = string
+}
+
+variable "resource_group_name" {
+  type = string
+}
+
+variable "vm_name" {
+  type    = string
+  default = "matebox"
+}
+
+variable "vm_size" {
+  type    = string
+  default = "Standard_B1s"
+}
+
+variable "subnet_id" {
+  type = string
+}
+
+variable "public_ip_id" {
+  type = string
+}
+
+variable "ssh_key_public" {
+  type = string
```

`ssh_key_public` is declared but not marked sensitive and lacks validation. I recommend adding `sensitive = true` and a `validation` block to ensure it is not empty so `terraform apply` fails early if no key was provided. Also add a `description`. This helps with security and correctness (checklist item 2.5).